src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -6422,4 +6789,400 @@ public static class D
+            class C
+            {
+                static void M() => throw null!;
+
+                static void Main()
+                {
+                    M();
+                }
+            }
+            """, "Program.cs", RegularWithInterceptors);
+
+        var comp = CreateCompilation(source);
+        var model = comp.GetSemanticModel(source);
+        var node = source.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().Single();
+        var locationSpecifier = model.GetInterceptableLocation(node)!;
+
+        var interceptors = CSharpTestSource.Parse($$"""
+            using System;
+
+            static class Interceptors
+            {
+                {{locationSpecifier.GetInterceptsLocationAttributeSyntax()}}
+                public static void M1() => Console.Write(1);
+            }
+            """, "Interceptors.cs", RegularWithInterceptors);
+
+        var verifier = CompileAndVerify([source, interceptors, CSharpTestSource.Parse(s_attributesSource.text, s_attributesSource.path, RegularWithInterceptors)], expectedOutput: "1");
+        verifier.VerifyDiagnostics();
+
+        // again, but using the accessors for specifically retrieving the individual attribute arguments
+        interceptors = CSharpTestSource.Parse($$"""
+            using System;
+            using System.Runtime.CompilerServices;
+
+            static class Interceptors
+            {
+                [InterceptsLocation({{locationSpecifier!.Version}}, "{{locationSpecifier.Data}}")]
+                public static void M1() => Console.Write(1);
+            }
+            """, "Interceptors.cs", RegularWithInterceptors);
+
+        verifier = CompileAndVerify([source, interceptors, CSharpTestSource.Parse(s_attributesSource.text, s_attributesSource.path, RegularWithInterceptors)], expectedOutput: "1");
+        verifier.VerifyDiagnostics();
+    }
+
+    [Fact]
+    public void Checksum_02()
+    {
+        var tree = CSharpTestSource.Parse("""
+            class C
+            {
+                static void M() => throw null!;
+
+                static void Main()
+                {
+                    M();
+                }
+            }
+            """.NormalizeLineEndings(), "path/to/Program.cs", RegularWithInterceptors);
+
+        var comp = CreateCompilation(tree);
+        var model = comp.GetSemanticModel(tree);
+        var node = tree.GetRoot().DescendantNodes().OfType<InvocationExpressionSyntax>().Single();
+        var locationSpecifier = model.GetInterceptableLocation(node)!;
+
+        // Verify behaviors of the public APIs.
+        Assert.Equal("path/to/Program.cs(7,9)", locationSpecifier.GetDisplayLocation());
+        Assert.Equal(1, locationSpecifier.Version);
+        Assert.Equal(locationSpecifier, locationSpecifier);
+
+        Assert.NotSame(locationSpecifier, model.GetInterceptableLocation(node));
+        Assert.Equal(locationSpecifier, model.GetInterceptableLocation(node));
```

Comment:
>Assert.Equal(locationSpecifier, model.GetInterceptableLocation(node)); [](http://example.com/codeflow?start=8&length=70)

Is this the only code path that tests `InterceptableLocation1.Equals()`? If so, consider testing other cases:
- `locationSpecifier.Equals(null)`
- `locationSpecifier.Equals(other)`